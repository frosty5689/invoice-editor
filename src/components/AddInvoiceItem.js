import React from 'react';
import { connect } from 'react-redux';
import { addLineItem } from '../redux/actions';

export class AddInvoiceItem extends React.Component {
  handleAddInvoiceItem() {
    this.props.addLineItem({name: '', quantity: 1, price: 0.00});
  };

  render() {
    return (
      <button onClick={() => this.handleAddInvoiceItem()}>New Item</button>
    );
  }
}

export default connect(
  null,
  { addLineItem }
)(AddInvoiceItem);