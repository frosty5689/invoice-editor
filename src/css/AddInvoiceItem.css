.Invoice-add-item {
  border: 0;
  padding: 0;
  background-color: transparent;
  color:lightgreen;
  cursor: pointer;
}
.Invoice-add-item:hover {
  color: lime;
}

.Invoice-add-item:focus {
  outline: 0;
}