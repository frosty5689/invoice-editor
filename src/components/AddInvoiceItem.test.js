import React from 'react';
import { AddInvoiceItem } from './AddInvoiceItem';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

Enzyme.configure({ adapter: new Adapter() });

describe('InvoiceItem component', () => {

  it('AddInvoiceItem should render correctly', () => {
    const component = renderer.create(
      <AddInvoiceItem />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AddInvoiceItem add calls addLineItemFn', () => {

    //addLineItem
    const mockAddLineItemFn = jest.fn();
    const component = shallow(
      <AddInvoiceItem addLineItem={mockAddLineItemFn}/>,
    );

    component.find('button').simulate('click');
    expect(mockAddLineItemFn.mock.calls.length).toBe(1);
    expect(mockAddLineItemFn.mock.calls[0][0].name).toBe('');
    expect(mockAddLineItemFn.mock.calls[0][0].quantity).toBe(1);
    expect(mockAddLineItemFn.mock.calls[0][0].price).toBe(0.00);
  });
});