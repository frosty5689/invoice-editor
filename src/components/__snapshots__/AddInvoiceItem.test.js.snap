// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceItem component AddInvoiceItem should render correctly 1`] = `
<button
  onClick={[Function]}
>
  New Item
</button>
`;
